src/classes/RD2_DataMigration_BATCH.cls

```diff
@@ -94,60 +117,111 @@ public class RD2_DataMigration_BATCH implements Database.Batchable<SObject>, Sch
     }
 
     /**
-    * @description Batch Execute Method
+    * @description Converts legacy Recurring Donation into enhanced Recurring Donation format
+    * @param context Batchable context
+    * @param rds Recurring Donations
+    * @return void
     */
     public void execute(Database.BatchableContext context, List<npe03__Recurring_Donation__c> rds) {
-        Map<Id, List<Opportunity>> oppsByRdId = getRelatedOpportunities(rds);
+        setBatchJobId(context);
 
-        Savepoint sp = Database.setSavepoint();
+        Map<Id, List<Opportunity>> oppsByRdId = getRelatedOpportunities(rds);
+        List<npe03__Recurring_Donation__c> convertedRDs = new List<npe03__Recurring_Donation__c>();
 
         for (npe03__Recurring_Donation__c rd : rds) {
-            handleConversion(rd, oppsByRdId.get(rd.Id));
+            RD2_DataMigrationMapper mapper = new RD2_DataMigrationMapper(rd)
+                .withRelatedOpportunities(oppsByRdId.get(rd.Id));
+
+            if (mapper.isConverted()) {	
+                continue;	
+            }
+
+            try {
+                convertedRDs.add(mapper.convertToEnhancedRD());
+
+            } catch (Exception e) {
+                logException(e, rd);
```

Comment:
Though technically this works, it's effectively issuing DML inside of a for loop. I recommend modifying this slightly so that the logException method instantiates the Error__c instance and collects them into a List<> that is then passed to the ERR_Handler.processErrorsOnly() method at the end of the method - ideally after the call to handleDmlAndLogging() so that the rollback for dry run mode (when implemented) will not prevent errors from being logged.

Reply:
Good catch, thanks @force2b ! The change has been applied.